tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+        test_cases = [
+            {
+                "args": (self.user, "ban", "2020-02-26 09:20 (23 hours and 59 minutes)"),
+                "expected_output": {
+                    "description": INFRACTION_DESCRIPTION_TEMPLATE.format(**{
+                        "type": "Ban",
+                        "expires": "2020-02-26 09:20 (23 hours and 59 minutes)",
+                        "reason": "No reason provided."
+                    }),
+                    "icon_url": Icons.token_removed,
+                    "footer": INFRACTION_APPEAL_FOOTER,
+                },
+                "send_result": True,
+                "send_raise": None
+            },
+            {
+                "args": (self.user, "warning", None, "Test reason."),
+                "expected_output": {
+                    "description": INFRACTION_DESCRIPTION_TEMPLATE.format(**{
+                        "type": "Warning",
+                        "expires": "N/A",
+                        "reason": "Test reason."
+                    }),
+                    "icon_url": Icons.token_removed,
+                    "footer": Embed.Empty
+                },
+                "send_result": False,
+                "send_raise": Forbidden(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, "note", None, None, Icons.defcon_denied),
+                "expected_output": {
+                    "description": INFRACTION_DESCRIPTION_TEMPLATE.format(**{
+                        "type": "Note",
+                        "expires": "N/A",
+                        "reason": "No reason provided."
+                    }),
+                    "icon_url": Icons.defcon_denied,
+                    "footer": Embed.Empty
+                },
+                "send_result": False,
+                "send_raise": NotFound(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, "mute", "2020-02-26 09:20 (23 hours and 59 minutes)", "Test", Icons.defcon_denied),
+                "expected_output": {
+                    "description": INFRACTION_DESCRIPTION_TEMPLATE.format(**{
+                        "type": "Mute",
+                        "expires": "2020-02-26 09:20 (23 hours and 59 minutes)",
+                        "reason": "Test"
+                    }),
+                    "icon_url": Icons.defcon_denied,
+                    "footer": INFRACTION_APPEAL_FOOTER
+                },
+                "send_result": False,
+                "send_raise": HTTPException(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            send, send_raise = case["send_result"], case["send_raise"]
+
+            with self.subTest(args=args, expected=expected, send=send, send_raise=send_raise):
+                if send_raise:
+                    self.ctx.send.side_effect = send_raise
```

Comment:
What is the purpose of this? `notify_infraction()` never calls `send()` directly and `send_private_embed()` is mocked so it won't call `send()` either.